gnomad_qc/v4/resources/meta.py

```diff
@@ -1,35 +1,55 @@
 """Script containing metadata related resources."""
 from gnomad.resources.resource_utils import TableResource, VersionedTableResource
 
-from gnomad_qc.v4.resources.constants import CURRENT_VERSION
+from gnomad_qc.v4.resources.constants import CURRENT_VERSION, VERSIONS
 
 
-def _meta_root_path(version: str = CURRENT_VERSION) -> str:
+def _meta_root_path(version: str = CURRENT_VERSION, data_type: str = "exomes") -> str:
     """
     Retrieve the path to the root metadata directory.
 
-    :param version: gnomAD version
-    :return: String representation of the path to the root metadata directory
+    :param version: gnomAD version.
+    :param data_type: Data type ("exomes" or "genomes"). Default is "exomes".
+    :return: String representation of the path to the root metadata directory.
     """
-    return f"gs://gnomad/v{version}/metadata/exomes"
+    return f"gs://gnomad/v{version}/metadata/{data_type}"
 
 
-def meta_tsv_path(version: str = CURRENT_VERSION) -> str:
+def meta_tsv_path(version: str = CURRENT_VERSION, data_type: str = "exomes") -> str:
     """
     Get the path to the finalized sample metadata information after sample QC.
 
-    :param version: gnomAD version
-    :return: String path to the finalized metadata
+    :param version: gnomAD version.
+    :param data_type: Data type ("exomes" or "genomes"). Default is "exomes".
+    :return: String path to the finalized metadata TSV.
     """
-    return f"{_meta_root_path(version)}/gnomad.exomes.v{version}.metadata.tsv.gz"
+    return (
+        f"{_meta_root_path(version, data_type)}/gnomad.{data_type}."
+        f"v{version}.metadata.tsv.gz"
+    )
 
 
-_meta_versions = {
-    "4.0": TableResource(
-        path="gs://gnomad/v4.0/metadata/exomes/gnomad.exomes.v4.0.sample_qc_metadata.ht"
-    ),
-}
+def meta(data_type: str = "exomes") -> VersionedTableResource:
+    """
+    Get the gnomAD v4 sample QC meta VersionedTableResource.
+
+    :param data_type: Data type ("exomes" or "genomes"). Default is "exomes".
+    :return: gnomAD v4 sample meta VersionedTableResource.
+    """
+    return VersionedTableResource(
+        CURRENT_VERSION,
```

Comment:
I just realized that this function assumes we will have a sample meta for both exomes and genomes for each version in `VERSIONS`, but what if we update the exome meta in a minor release and don't update the genomes?

I realize this makes the function longer, but it's probably a good idea to have this function take in version as an argument (with the default set to `CURRENT_VERSION`), re-add a meta_versions dictionary that tracks the available sample meta versions for both data types, and does a check at the top of the function verifying that sample meta exists for the specified combination of version and data_type